src/lib/instruction-parser.ts

```diff
@@ -238,17 +253,37 @@ export class InstructionParser {
     for (const instruction of tx.meta.innerInstructions) {
       if (instruction.index === routeInfo.index) {
         innerInstructions = instruction.instructions;
-        instruction.instructions.forEach((instruction, index) => {
-          if (isSwapInstruction(instruction)) {
+        for (let index = 0; index < innerInstructions.length; index++) {
+          if (isSwapInstruction(innerInstructions[index])) {
             const routePlanIndex = swaps.length;
+            const swapProgramId = innerInstructions[index].programId;
+            const routePlan = routeInfo.routePlan[routePlanIndex];
+            const swapIxName = Object.keys(routePlan.swap)[0];
+            if (CUSTOM_SWAPS.includes(swapIxName)) {
+              const { inAccount, outAccount, skipToIndex } = this.getCustomSwap(
+                innerInstructions,
+                index,
+                swapProgramId,
+                swapIxName,
+                routePlan.swap
+              );
+              swaps.push({
+                instructionIndex: index,
+                isAsk: true, // swap direction is already determined at this point
+                inAccount,
+                outAccount,
+              });
+              index = skipToIndex;
```

Comment:
why do you need this?

Reply:
Custom swaps call two program instructions to complete whole swap. This is used to skip the second instruction from being considered as a new swap